src/libs/actions/IOU.ts

```diff
@@ -20,37 +21,136 @@ import Permissions from '@libs/Permissions';
+type OptimisticPolicyRecentlyUsedTags = Record<string, string[]>;
+
+type MoneyRequestRoute = StackScreenProps<
+    MoneyRequestNavigatorParamList,
+    typeof SCREENS.MONEY_REQUEST.CATEGORY | typeof SCREENS.MONEY_REQUEST.TAG | typeof SCREENS.MONEY_REQUEST.CONFIRMATION
+>['route'];
+
+type IOURequestType = ValueOf<typeof CONST.IOU.REQUEST_TYPE>;
+
+type PaymentMethodType = DeepValueOf<typeof CONST.IOU.PAYMENT_TYPE>;
+
+type OneOnOneIOUReport = OnyxTypes.Report | undefined | null;
+
+type MoneyRequestInformation = {
+    payerAccountID: number;
+    payerEmail: string;
+    iouReport: OnyxTypes.Report;
+    chatReport: OnyxTypes.Report;
+    transaction: OnyxTypes.Transaction;
+    iouAction: OptimisticIOUReportAction;
+    createdChatReportActionID: string;
+    createdIOUReportActionID: string;
+    reportPreviewAction: OnyxTypes.ReportAction;
+    onyxData: OnyxData;
+};
+
+type SplitData = {
+    chatReportID: string;
+    transactionID: string;
+    reportActionID: string;
+    policyID?: string;
+    createdReportActionID?: string;
+};
+
+type SplitsAndOnyxData = {
+    splitData: SplitData;
+    splits: Split[];
+    onyxData: OnyxData;
+};
+
+type UpdateMoneyRequestParams = Partial<TransactionDetails> & {
+    reportID?: string;
+    transactionID: string;
+    reportActionID?: string;
+};
+
+type UpdateMoneyRequestData = {
+    params: UpdateMoneyRequestParams;
+    onyxData: OnyxData;
+};
+
+type PayMoneyRequestParams = {
+    iouReportID: string;
+    chatReportID: string;
+    reportActionID: string;
+    paymentMethodType: PaymentMethodType;
+};
+
+type PayMoneyRequestData = {
+    params: PayMoneyRequestParams;
+    optimisticData: OnyxUpdate[];
+    successData: OnyxUpdate[];
+    failureData: OnyxUpdate[];
+};
+
+type SendMoneyParams = {
+    iouReportID: string;
+    chatReportID: string;
+    reportActionID: string;
+    paymentMethodType: PaymentMethodType;
+    transactionID: string;
+    newIOUReportDetails: string;
+    createdReportActionID: string;
+    reportPreviewReportActionID: string;
+};
+
+type SendMoneyParamsData = {
+    params: SendMoneyParams;
+    optimisticData: OnyxUpdate[];
+    successData: OnyxUpdate[];
+    failureData: OnyxUpdate[];
+};
+
+let betas: OnyxTypes.Beta[] = [];
 Onyx.connect({
     key: ONYXKEYS.BETAS,
-    callback: (val) => (betas = val || []),
+    callback: (val) => (betas = val ?? []),
 });
 
-let allPersonalDetails;
+let allPersonalDetails: OnyxTypes.PersonalDetailsList = {};
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
     callback: (val) => {
-        allPersonalDetails = val || {};
+        allPersonalDetails = val ?? {};
     },
 });
 
-let allReports;
+let allReports: OnyxCollection<OnyxTypes.Report> = null;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (val) => (allReports = val),
 });
 
-let allTransactions;
+let allTransactions: Record<string, OnyxEntry<OnyxTypes.Transaction>> = {};
```

Comment:
@fabioh8010 I didn't type it as an `OnyxCollection`, because `OnyxCollection` includes `null`, and in this case `allTransactions` won't be null, since it has a [check](https://github.com/Expensify/App/pull/34092/files#diff-9c07b44e092fd3d975806649be46d76b75a377c16e32ed86597b630a84c290d0R158)

Reply:
@VickyStash what about `NonNullable<OnyxCollection<PersonalDetails>>`? Same for other comments.